src/main/java/com/mineboundteam/minebound/magic/UtilitySpells/ShieldUtilitySpell.java

```diff
@@ -69,29 +71,42 @@ public static void onPlayerTick(TickEvent.PlayerTickEvent event) {
         if (event.side.isServer() && event.phase == TickEvent.Phase.START) {
             ItemStack spellStack = getHighestEquippedSpellOfType(ShieldUtilitySpell.class, event.player);
             if (spellStack != null) {
-                ShieldUtilitySpell spell = (ShieldUtilitySpell) spellStack.getItem();
-                if (spellStack.hasTag()) {
-                    CompoundTag spellTag = spellStack.getOrCreateTag();
-                    int cooldown = spellTag.getInt("minebound.shield_utility.cooldown");
-                    if (cooldown > 0) {
-                        spellTag.putInt("minebound.shield_utility.hits_remaining", spellTag.getInt("minebound.shield_utility.hits_remaining"));
-                        spellTag.putInt("minebound.shield_utility.cooldown", cooldown - 1);
-                    } else if (cooldown == 0) {
-                        spellTag.putInt("minebound.shield_utility.hits_remaining", spell.totalHits);
-                        spellTag.putInt("minebound.shield_utility.cooldown", -1);
-                    }
+                CompoundTag spellTag = spellStack.getOrCreateTagElement("minebound.shield_utility");
+                int cooldown = spellTag.getInt("cooldown");
+                int shieldMax = getShieldCount(event.player);
+                float hitsRemaining = spellTag.getFloat("hits_remaining");
+
+                if (cooldown > 0) {
+                    spellTag.putInt("cooldown", cooldown - 1);
+                } else if (hitsRemaining < shieldMax) {
+                    spellTag.putFloat("hits_remaining", hitsRemaining+0.5f);
+                    spellTag.putInt("cooldown", -1);
                 }
+                
+                if(hitsRemaining > shieldMax)
+                    spellTag.putFloat("hits_remaining", shieldMax);
```

Comment:
Could simplify this by just doing `Math.min(hitsRemaining + 0.5f, shieldMax)` on line 82.

Reply:
Fixed